fetchcode/package.py

```diff
@@ -0,0 +1,330 @@
+        primary_language=primary_language,
+        **purl.to_dict(),
+    )
+    release_url = f"{api_url}/releases"
+    releases = get_response(release_url)
+    for release in releases:
+        version = release.get("name")
+        version_purl = PackageURL(
+            type=purl.type, namespace=namespace, name=name, version=version
+        )
+        download_url = release.get("tarball_url")
+        code_view_url = f"{github_url}/{namespace}/{name}/tree/{version}"
+        version_vcs_url = f"{vcs_url}@{version}"
+        yield Package(
+            homepage_url=homepage_url,
+            vcs_url=version_vcs_url,
+            api_url=api_url,
+            bug_tracking_url=bug_tracking_url,
+            code_view_url=code_view_url,
+            declared_license=declared_license,
+            primary_language=primary_language,
+            download_url=download_url,
+            **version_purl.to_dict(),
+        )
+
+
+@router.route("pkg:bitbucket/.*")
+def get_bitbucket_data_from_purl(purl):
+    """
+    Generate `Package` object from the `purl` string of bitbucket type
+    """
+    purl = PackageURL.from_string(purl)
+    name = purl.name
+    namespace = purl.namespace
+    base_path = "https://api.bitbucket.org/2.0/repositories"
+    api_url = f"{base_path}/{namespace}/{name}"
+    response = get_response(api_url)
+    bitbucket_url = "https://bitbucket.org"
+    bug_tracking_url = f"{bitbucket_url}/{namespace}/{name}/issues"
+    code_view_url = f"{bitbucket_url}/{namespace}/{name}"
+    yield Package(
+        api_url=api_url,
+        bug_tracking_url=bug_tracking_url,
+        code_view_url=code_view_url,
+        **purl.to_dict(),
+    )
+    links = response.get("links") or {}
+    tags_url = links.get("tags") or {}
+    tags_url = tags_url.get("href")
+    if not tags_url:
+        return []
+    tags_data = get_response(tags_url)
+    tags = tags_data.get("values") or {}
+    for tag in tags:
+        version = tag.get("name") or ""
+        version_purl = PackageURL(
+            type=purl.type, namespace=namespace, name=name, version=version
+        )
+        download_url = (
+            f"{base_path}/{namespace}/{name}/downloads/{name}-{version}.tar.gz"
+        )
+        code_view_url = f"{bitbucket_url}/{namespace}/{name}/src/{version}"
+        yield Package(
+            api_url=api_url,
+            bug_tracking_url=bug_tracking_url,
+            code_view_url=code_view_url,
+            download_url=download_url,
+            **version_purl.to_dict(),
+        )
+
+
+@router.route("pkg:rubygems/.*")
+def get_rubygems_data_from_purl(purl):
+    """
+    Generate `Package` object from the `purl` string of rubygems type
+    """
+    purl = PackageURL.from_string(purl)
+    name = purl.name
+    api_url = f"https://rubygems.org/api/v1/gems/{name}.json"
+    response = get_response(api_url)
+    declared_license = response.get("licenses") or [None]
+    declared_license = declared_license[0]
```

Comment:
You cannot ignore other licenses. Return a list.
For instance:
https://rubygems.org/api/v1/gems/cairo.json